### PR TITLE
feat(action-bar-icon-interface): make `icon` field required when `iconOnly` is true

### PR DESCRIPTION
**Pull request by LawrenceBorst**
Fixes Lundalogik/crm-feature#3452

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/action-bar/action-bar.types.ts [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-     * Does not affect the items that are overflown into the overflow menu.
-     */
-    iconOnly?: boolean;
+/**
+ * Renders the button in the action bar without their labels.
+ * Does not affect the items that are overflown into the overflow menu.
+ */
+export type ActionBarItem = ABIWithIcon | ABIWithoutIcon;
+
+interface ABIWithIcon extends MenuItem {
```
Comment (jgroth) [inline src/components/action-bar/action-bar.types.ts:9]: Please don't use abbreviations :smile:

File: src/components/action-bar/action-bar.types.ts [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
+    iconOnly: true;
+    icon: string;
+}
+
+interface ABIWithoutIcon extends MenuItem {
```
Comment (civing) [inline src/components/action-bar/action-bar.types.ts:14]: I think you should name it `ActionBarItemWithLabel`. An action bar with `iconOnly: False` could still show an icon if set.

**Review by civing — COMMENTED** [review]
Really neat! Tested it and it worked really good. Also, I agree with @jgroth about not having abbreviations in the names.

File: src/components/action-bar/action-bar.types.ts [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
-     * Does not affect the items that are overflown into the overflow menu.
-     */
-    iconOnly?: boolean;
+/**
+ * Renders the button in the action bar without their labels.
+ * Does not affect the items that are overflown into the overflow menu.
+ */
+export type ActionBarItem = ActionBarItemWithLabel | ActionBarItemWithoutLabel;
+
+interface ActionBarItemWithLabel extends MenuItem {
```
Comment (civing) [inline src/components/action-bar/action-bar.types.ts:9]: Now you swapped them again 🤣 
```suggestion
interface ActionBarItemOnlyIcon extends MenuItem {
```

File: src/components/action-bar/action-bar.types.ts [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
+    iconOnly: true;
+    icon: string;
+}
+
+interface ActionBarItemWithoutLabel extends MenuItem {
```
Comment (civing) [inline src/components/action-bar/action-bar.types.ts:14]: This one is with label
```suggestion
interface ActionBarItemWithLabel extends MenuItem {
```

Comment (LawrenceBorst) [reply src/components/action-bar/action-bar.types.ts:14]: Woops autopilot 😅

**Review by civing — APPROVED** [review]
Awesome! 🥳

**Pull request merged**